src/Blocks/BlockCrops.h

```diff
@@ -23,7 +23,30 @@ class cBlockCropsHandler :
 	{
 		cFastRandom rand;
 
-		if (a_Meta == 0x7)
+		// Beetroots have three stages before fully grown
+		if (a_Meta == 0x3)
+		{
+			// Is fully grown, drop the entire produce:
+			switch (m_BlockType)
+			{
+				case E_BLOCK_BEETROOTS:
+				{
+					char SeedCount = static_cast<char>(1 + (rand.NextInt(3) + rand.NextInt(3)) / 2);
+					a_Pickups.push_back(cItem(E_ITEM_BEETROOT_SEEDS, SeedCount, 0));
+					char BeetrootCount = static_cast<char>(1 + (rand.NextInt(3) + rand.NextInt(3)) / 2);
+					a_Pickups.push_back(cItem(E_ITEM_BEETROOT, BeetrootCount, 0));  // [1 .. 3] with high preference of 2
+					break;
+				}
+				default:
+				{
+					ASSERT(!"Unhandled block type");
+					break;
+				}
```

Comment:
Sorry, I don't quite understand what to do. I'm not familiar with the terminology or programming language.